autonomy/deploy/base.py

```diff
@@ -626,10 +610,12 @@ def generate_common_vars(self, agent_n: int) -> Dict:
             ENV_VAR_AEA_AGENT: self.service.agent,
             ENV_VAR_LOG_LEVEL: self.log_level,
         }
-
-        if self.private_keys_password is not None:
-            agent_vars[ENV_VAR_AEA_PASSWORD] = self.private_keys_password
-
+        if self.deplopyment_type == DOCKER_COMPOSE_DEPLOYMENT:
+            agent_vars[ENV_VAR_AEA_PASSWORD] = "$OPEN_AUTONOMY_PRIVATE_KEY_PASSWORD"
+        else:
+            agent_vars[ENV_VAR_AEA_PASSWORD] = os.environ.get(
+                "OPEN_AUTONOMY_PRIVATE_KEY_PASSWORD", ""
+            )
```

Comment:
Unfortunately the kubernetes does not provide such functionalities so we'll have to store them the old way